san/transform.py

```diff
@@ -0,0 +1,30 @@
+import operator
+from functools import reduce
+
+QUERY_PATH_MAP = {
+    'eth_top_transactions': ['ethTopTransactions']
+}
+
+def path_to_data(idx, query, data):
+    return reduce(operator.getitem, ['query_'+str(idx),]+QUERY_PATH_MAP[query], data)
+    """
+    With this function we jump straight onto the key from the dataframe, that we want and start from there
+    """
+
+def add_transform(function):
+    def decorator(idx, query, data):
```

Comment:
My original idea when suggesting decorators was that every transform function will be decorated as to show what query it transforms. Though the currently implemented way is with dynamically choosing the transform function based on its string name.

In this case the decorator could be removed as it's not doing any work - we can move the `transform_query` function here, call it just `transform_result` (check the other comment) and leave the definition of `decorator` as its implementation (+ call the `convert` on the result instead of `function`)

Reply:
Makes sense. Decorator is not needed. The solution with checking globals whether a transform function exists is sufficient. For the name: I think `transform_query_result` - cause the transformation is based on the query name

Reply:
Removed the decorator and renamed the function, it really fits it more than just transform_query